src/main/java/ai/starlake/transpiler/schema/JdbcMetaData.java

```diff
@@ -41,31 +44,48 @@
 import java.util.function.Function;
 import java.util.logging.Logger;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonIncludeProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import ai.starlake.transpiler.schema.JdbcUtils.DatabaseSpecific;
+
 /**
  * The type Jdbc metadata.
  */
+@JsonPropertyOrder({"databaseType","currentCatalog","currentSchema","catalogSeparator","catalogs"})
+@JsonIncludeProperties({"currentCatalog","currentSchema","catalogSeparator","databaseType","catalogs"})
+@JsonIgnoreProperties(ignoreUnknown = true)
 @SuppressWarnings({"PMD.CyclomaticComplexity"})
 public final class JdbcMetaData implements DatabaseMetaData {
   public final static Logger LOGGER = Logger.getLogger(JdbcMetaData.class.getName());
   public static final Map<Integer, String> SQL_TYPE_NAME_MAP = new HashMap<>();
 
-  private final CaseInsensitiveLinkedHashMap<JdbcCatalog> catalogs =
+  private CaseInsensitiveLinkedHashMap<JdbcCatalog> catalogs =
       new CaseInsensitiveLinkedHashMap<>();
-  private final String currentCatalogName;
```

Comment:
I think this better shall be final. We are not sure what happens when you change those during runtime/lifetime of the object. I would prefer rather immutable objects.

Reply:
The `LinkedHashMap catalogs` could remain final, `currentCatalogName` not - due to the added (de)serialization. 
Plus, possibly supporting changing of current catalog & current schema could support doing the statement analysis under different context on larger DB catalogs where there are multiple catalogs and/or schemas where tables&columns referenced by SQL query can be different.
This again supports the idea that the DB catalog is extracted once and then potentially analyzed multiple times with different context set.